### PR TITLE
KTOR-4877 Fix UDP already connected

### PR DESCRIPTION
**Pull request by e5l**
Fix [KTOR-4877](https://youtrack.jetbrains.com/issue/KTOR-4877/POSIX-error-56-Socket-is-already-connected-error-when-a-socket-is-connection-mode-on-Darwin-targets)

### CODE REVIEW

File: ktor-network/jvmAndNix/test/io/ktor/network/sockets/tests/UDPSocketTest.kt [hunk 1]
```diff
@@ -213,6 +215,18 @@ class UDPSocketTest {
                 }
             }
     }
+
+    @Test
+    fun testUdpConnect() = testSockets { selector ->
```
Comment (rsinukov) [inline ktor-network/jvmAndNix/test/io/ktor/network/sockets/tests/UDPSocketTest.kt:220]: this test failed twice in different builds and succeed it others. can you check it please?

**Review by rsinukov — CHANGES_REQUESTED** [review]

File: ktor-network/jvmAndNix/test/io/ktor/network/sockets/tests/UDPSocketTest.kt (hunk 1 above)
Comment (e5l) [reply ktor-network/jvmAndNix/test/io/ktor/network/sockets/tests/UDPSocketTest.kt:220]: it looks like flaky, will take a look

Comment (e5l) [reply ktor-network/jvmAndNix/test/io/ktor/network/sockets/tests/UDPSocketTest.kt:220]: It looks like a different behavior in win API, will take a look

Comment (e5l) [reply ktor-network/jvmAndNix/test/io/ktor/network/sockets/tests/UDPSocketTest.kt:220]: fixed: you cannot bind and connect single hostname and port on win

**Review by rsinukov — APPROVED** [review]

**Review by rsinukov — CHANGES_REQUESTED** [review]
Still failing on CI

File: ktor-network/nix/src/io/ktor/network/sockets/DatagramSendChannel.kt [hunk 2]
```diff
@@ -97,30 +92,53 @@ internal class DatagramSendChannel(
...
             sendImpl(element)
         }
     }
 
-    @OptIn(UnsafeNumber::class)
-    private tailrec suspend fun sendImpl(
-        datagram: Datagram,
-        bytes: ByteArray = datagram.packet.readBytes()
-    ) {
+    private fun sento(datagram: Datagram, bytes: ByteArray): Int {
```
Comment (rsinukov) [inline ktor-network/nix/src/io/ktor/network/sockets/DatagramSendChannel.kt:106]: `sento` -> `sendto`

**Review by rsinukov — APPROVED** [review]

**Pull request merged**